mypy/stubgen.py

```diff
@@ -90,28 +90,63 @@
 from mypy.build import build
 from mypy.errors import CompileError, Errors
 from mypy.traverser import has_return_statement
+from mypy.moduleinspect import ModuleInspect
 
 
 # Common ways of naming package containing vendored modules.
 VENDOR_PACKAGES = [
     'packages',
     'vendor',
     'vendored',
+    '_vendor',
+    '_vendored_packages',
 ]
 
 # Avoid some file names that are unnecessary or likely to cause trouble (\n for end of path).
 BLACKLIST = [
     '/six.py\n',  # Likely vendored six; too dynamic for us to handle
     '/vendored/',  # Vendored packages
     '/vendor/',  # Vendored packages
+    '/_vendor/',
+    '/_vendored_packages/',
 ]
 
 # Special-cased names that are implicitly exported from the stub (from m import y as y).
 EXTRA_EXPORTED = {
     'pyasn1_modules.rfc2437.univ',
     'pyasn1_modules.rfc2459.char',
     'pyasn1_modules.rfc2459.univ',
-    'elasticsearch.client.utils._make_path',
+}
+
+# These names should be omitted from generated stubs.
+IGNORED_DUNDERS = {
+    '__all__',
+    '__author__',
+    '__version__',
+    '__about__',
+    '__copyright__',
+    '__email__',
+    '__license__',
+    '__summary__',
+    '__title__',
+    '__uri__',
+    '__str__',
+    '__repr__',
+    '__getstate__',
+    '__setstate__',
+    '__slots__',
+}
+
+# These methods are expected to always return a non-trivial value.
+METHODS_WITH_RETURN_VALUE = {
+    '__ne__',
+    '__eq__',
+    '__lt__',
+    '__le__',
+    '__gt__',
+    '__ge__',
+    '__hash__',
+    '__iter__',
 }
```

Comment:
Not really important but I think all these top-level constants can be made `Final`.